### DISCUSSION TITLE
The negative values of train dice.

### DISCUSSION THREAD

**Discussion by vivekmodgill** [Project-MONAI/MONAI#4469] [Q&A] [answered]
Hi,
We tried to run a sub-sample with medical images to train a model to detect lesions and the train dice is negative! Can anyone suggest where are we going wrong?

![Capture](https://user-images.githubusercontent.com/27670197/172468739-2685f090-7d10-41ff-9719-264351b57f84.PNG)

@alcarpo @ericspod

**Comment by ericspod** [comment, marked as answer]
This happens for example if you use the binary formulation for dice with multiclass inputs, since the classes after the first two have larger values the subtraction can work out to be negative. It could also be that your prediction isn't being activated with sigmoid if binary or with softmax if multi-class, these values must be activated before going into the loss function. There are questions on dice that keep coming up along these lines so I wrote a notebook on the loss and metric that explains some of the subtle issues: https://github.com/Project-MONAI/tutorials/blob/main/modules/dice_loss_metric_notes.ipynb I hope that helps.
